design/authenticators/authenticator_management_improvement.md

```diff
@@ -0,0 +1,248 @@
+
+conjur config authenticator add *authenticator-type/service-id* [*parameters*]
+
+#### Example
+
+```shell script
+$ conjur config authenticator add authn-oidc/my-idp --provider-uri my-uri --id-token-user-property prop
+```
+
+### Delete Authenticator
+
+#### Functionality
+
+Deletes an existing authenticator in Conjur. Deletes the authenticator policy and deletes the authenticator from the enabled/disabled authenticators list.
+
+#### Synopsis
+
+conjur config authenticator delete *authenticator-type/service-id*
+
+#### Example
+
+```shell script
+$ conjur config authenticator delete authn-oidc/my-idp
+```
+
+### Update Authenticator
+
+#### Functionality
+
+Updates an existing authenticator in Conjur. Updates the authenticator variables.
+
+#### Synopsis
+
+conjur config authenticator update *authenticator-type/service-id* [*parameters*]
+
+#### Example
+
+```shell script
+$ conjur config authenticator update authn-oidc/my-idp --provider-uri new-uri
+```
+
+### Enable Authenticator
+
+#### Functionality
+
+Enables an authenticator in the Conjur database (can be overriden by environment variable).
+
+#### Synopsis
+
+conjur config authenticator enable *authenticator-type/service-id*
+
+#### Example
+
+```shell script
+$ conjur config authenticator enable authn-oidc/my-idp
+```
+
+### Disable Authenticator
+
+#### Functionality
+
+Disables an authenticator in the Conjur database (can be overriden by environment variable).
+
+#### Synopsis
+
+conjur config authenticator disable *authenticator-type/service-id*
+
+#### Example
+
+```shell script
+$ conjur config authenticator disable authn-oidc/my-idp
+```
+
+## Backwards Compatibility
+
+- The authenticators keep their same structure in Conjur. Users could still manage them in the way it's done today.  
+- These CLI commands are uniting policy loads and variable updates into a single step, therefore running the commands requires permissions to both load a policy under the `conjur` policy and to update the variables in that policy.
+
+## Performance
+
+The proposal only simplifies the steps, not adding or changing them, therefore the performance remain the same.
+
+## Security
+
+No security implications. These new CLI commands perform the same action, with an easier experience.
+
+## Documentation
+
+We will need to update the docs of every authenticator, to specify how to use the new CLI commands. In addition, we will need to add documentation for enabling/disabling authenticators.
+
+## Implementation Options
+
+### Alternative 1 - Server Side
+
+The new functionality will be developed on the Conjur side. New APIs will be exposed for this new functionality. For example, in order to add an authenticator, the client will call a `POST` request to `/{authenticator-type}/{service-id}/{account}`. Example:
+
+```shell script
+curl --request POST \
+  --data '{"providerUri": "my-uri", "id-token-user-property": "prop"}' \
+  https://conjur.mycompany.net/authn-oidc/my-authenticator/my-account
```

Comment:
The above syntax raises the question if authenticators should be allowed to span Conjur accounts. Conjur accounts were originally intended to allow separation between groups (ex. staging/production, or marketing/infrastructure/etc.).

Reply:
Why does this syntax raise this question? Currently authenticators don't span Conjur accounts and in this example, the specific authenticator under the specific account are specified in the URL. To compare, to authenticate with an authenticator you would call `https://conjur.mycompany.net/authn-oidc/my-authenticator/my-account/authenticate`

Reply:
The above is missing the account scope in the URL.